### ISSUE TITLE
build.sh fails to properly set qmake path

### ISSUE THREAD

**Issue by pdurusau** [radareorg/cutter#877]
Ubuntu 18.04
Cutter: 1.7.2

**Describe the bug**

I have qmake but not qmake-qt5 installed. The build.sh script fails to return the path to gmake.

**To Reproduce**
1. Remove qmake-qt5
2. Run build.sh
3. Even if gmake is present on system, build.sh will not report it
4. Error is that gmake isn't found.

**Expected behavior**
Expected gmake to be found with (which gmake) and reported to build.sh

**Additional context**

The relevant part of build.sh reads as follows:

# Check if either qmake or qmake-qt5 is available
qmakepath=$(which qmake-qt5)
if [ -z "$qmakepath" ]; then
         qmakepath=$(which qmake)
fi

Notice the first setting of qmakepath occurs OUTSIDE the if statement. The if statement spawns a subshell and so the result of "qmakepath=$(which qmake)" is not communicated to the script.

If qmake-qt5 isn't present but gmake is, the build script fails.

I was able to build but only by putting gmake outside the if statement. 

Thanks for all the great work! I suspect the qmake-qt5 should be within the if statement and if it succeeds, then exporting that value, followed by an if for qmake and then an export statement, but I'm no shell jockey.

**Comment by xarkes** [comment]
> The if statement spawns a subshell and so the result of "qmakepath=$(which qmake)" is not communicated to the script.

I think this statement is wrong? At least, it works here in my machine:

```diff
 # Check if either qmake or qmake-qt5 is available
-qmakepath=$(which qmake-qt5)
+qmakepath=$(which something_not_in_PATH 2>/dev/null)
 if [ -z "$qmakepath" ]; then
-       qmakepath=$(which qmake)
+       qmakepath=$(which qmake 2>/dev/null)
 fi
 if [ -z "$qmakepath" ]; then
        echo "You need qmake to build Cutter."
        echo "Please make sure qmake is in your PATH environment variable."
        exit 1
 fi
+echo "Found '${qmakepath}'"
```
![2018-10-26-084324_730x95_scrot](https://user-images.githubusercontent.com/10772605/47549069-386f2880-d8fb-11e8-9385-3ee8930b944e.png)

**Comment by pdurusau** [comment]
User error! I won't waste your time with how I discovered this error but I was sudo ./build.sh as opposed to simply ./build.sh. 

Sudo versus the default user in the VM return very different results for "which," causing the build script to fail. After seeing this result with a test script, I removed Cutter and re-installed from a fresh clone to verify my results. (Success!)

Apologies for the distraction.
